File: FormularioClase4 C#/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace FormularioClase4CSharp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            var result= MessageBox.Show("¿Está seguro que desea salir?", "Salir del sistema", MessageBoxButtons.YesNo, MessageBoxIcon.Exclamation);
            if (result == System.Windows.Forms.DialogResult.Yes)
            {
                this.Close();
            }  
         }

        private void button2_Click(object sender, EventArgs e)
        {
            Form Cli = new Clientes();
            Cli.BackColor = Color.Red;
            Cli.ShowDialog();
        }

        private void button3_Click(object sender, EventArgs e)
        {
            Form Pro = new Proveedores();
            Pro.BackColor = Color.Green;
            Pro.ShowDialog();
        }

        private void button4_Click(object sender, EventArgs e)
        {
            Form List = new Listado();
            List.BackColor = Color.Blue;
            List.ShowDialog();
        }

        private void listadoToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Form List = new Listado();
            List.BackColor = Color.Blue;
            List.ShowDialog();
        }

        private void clientesToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Form Cli = new Clientes();
            Cli.BackColor = Color.Red;
            Cli.ShowDialog();
        }

        private void proveedoresToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Form Pro = new Proveedores();
            Pro.BackColor = Color.Green;
            Pro.ShowDialog();
        }
    }
    }

File: FormulariosClase4 VB/Form1.vb
Public Class Form1
    Private Sub bSalir_Click(sender As Object, e As EventArgs) Handles bSalir.Click
        If MessageBox.Show("¿Está seguro que desea salir?", "Salir del sistema", MessageBoxButtons.YesNo, MessageBoxIcon.Exclamation, MessageBoxDefaultButton.Button2) = DialogResult.Yes Then End
    End Sub

    Private Sub bClientes_Click(sender As Object, e As EventArgs) Handles bClientes.Click
        Dim cli As New Clientes
        cli.ShowDialog()
        cli.BackColor = Color.Red
        cli.ShowDialog()
    End Sub

    Private Sub bProveedores_Click(sender As Object, e As EventArgs) Handles bProveedores.Click
        Dim Pro As New Proveedores
        Pro.ShowDialog()
        Pro.BackColor = Color.Green
        Pro.ShowDialog()
    End Sub

    Private Sub bListados_Click(sender As Object, e As EventArgs) Handles bListados.Click
        Dim List As New Listados
        List.ShowDialog()
        List.BackColor = Color.Blue
        List.ShowDialog()
    End Sub

    Private Sub ClientesToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles ClientesToolStripMenuItem.Click
        Dim cli As New Clientes
        cli.ShowDialog()
        cli.BackColor = Color.Red
        cli.ShowDialog()
    End Sub

    Private Sub ProveedoresToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles ProveedoresToolStripMenuItem.Click
        Dim Pro As New Proveedores
        Pro.ShowDialog()
        Pro.BackColor = Color.Green
        Pro.ShowDialog()
    End Sub

    Private Sub ListadosToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles ListadosToolStripMenuItem.Click
        Dim List As New Listados
        List.ShowDialog()
        List.BackColor = Color.Blue
        List.ShowDialog()
    End Sub
End Class